### PR TITLE
Toggle Comments Inconsistent Behavior With comments.visible Setting

### LINKED ISSUE #167663: Toggle Comments Inconsistent Behavior With comments.visible Setting
Testing #167403

I followed the steps and did see the correct behavior when I set comments.visible to false.
If I do not reload the window after this setting, I can find comments: toggle editor commenting` and it will still change the behavior of the comments.

If I reload the window after editing the setting when I search for `comments: toggle editor commenting` in the command palette and `comments.visible: false` then that command does not show up in the command palette.

### PR DESCRIPTION
**Pull request by alexr00**
Fixes #167663

### CODE REVIEW

File: src/vs/workbench/contrib/comments/browser/commentService.ts [hunk 1]
```diff
@@ -16,9 +17,15 @@ import { ICellRange } from 'vs/workbench/contrib/notebook/common/notebookRange';
 import { IWorkbenchLayoutService } from 'vs/workbench/services/layout/browser/layoutService';
 import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
 import { COMMENTS_SECTION, ICommentsConfiguration } from 'vs/workbench/contrib/comments/common/commentsConfiguration';
+import { IContextKey, IContextKeyService, RawContextKey } from 'vs/platform/contextkey/common/contextkey';
 
 export const ICommentService = createDecorator<ICommentService>('commentService');
 
+export const WorkspaceHasCommenting = new RawContextKey<boolean>('workspaceHasCommenting', false, {
+	description: nls.localize('hasCommentingProvider', "Whether the open workspace has either comments or commenting ranges."),
```
Comment (Tyriar) [inline src/vs/workbench/contrib/comments/browser/commentService.ts:25]: Should this be "comment provider"?

**Review by Tyriar — COMMENTED** [review]
I'm not sure this meets the new severity bar for fixes after testing has occurred? Should we merge this for December instead?

File: src/vs/workbench/contrib/comments/browser/commentService.ts (hunk 1 above)
Comment (alexr00) [reply src/vs/workbench/contrib/comments/browser/commentService.ts:25]: Which part?

**Comment by alexr00** [comment]
> I'm not sure this meets the new severity bar for fixes after testing has occurred? Should we merge this for December instead?

This is a bug in a new feature that was discovered during testing. We are not doing candidates only at this time, so this meets the fix bar.

File: src/vs/workbench/contrib/comments/browser/commentService.ts (hunk 1 above)
Comment (Tyriar) [reply src/vs/workbench/contrib/comments/browser/commentService.ts:25]: Oh never mind, I see it's `CommentingRangeProvider` in the API. Did a search for commentprovider which showed results which is what my comment is based on.

**Review by Tyriar — APPROVED** [review]
I agree if it's going to be mentioned in the release notes, it looked like a minor bug just reading the report is all.

**Comment by alexr00** [comment]
The new feature is the `comments.visible` setting and the bug is between the interaction with this setting and the existing command "Comments: Toggle Editor Commenting". Part of the selling point for the setting is the way it interacts with the command (similar to View: Toggle Word Wrap).

**Pull request merged**